pkg/crinit/aggregated/aggregated_test.go

```diff
@@ -0,0 +1,515 @@
+		},
+		{
+			desc: "advertise port rejected if out of range",
+			initialOpts: &aggregatedClusterRegistryOptions{
+				apiServerServiceTypeString: string(v1.ServiceTypeNodePort),
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerNodePortPort: 100000}},
+			errExpected: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			err := validateOptions(tc.initialOpts)
+			if tc.errExpected {
+				if err == nil {
+					t.Error("expected error, but no error was returned")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("no error expected, but got %v", err)
+				}
+				if tc.finalOpts != nil {
+					if !reflect.DeepEqual(tc.initialOpts, tc.finalOpts) {
+						t.Errorf("unexpected output: got: %v, want: %v", tc.initialOpts, tc.finalOpts)
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestMarshalOptions(t *testing.T) {
+	testCases := []struct {
+		desc           string
+		overrideParams string
+		expectedMap    map[string]string
+		expectedErr    string
+	}{
+		{
+			desc:           "valid arguments",
+			overrideParams: "valid-format-param1=override1,valid-format-param2=override2",
+			expectedMap:    map[string]string{"valid-format-param1": "override1", "valid-format-param2": "override2"},
+			expectedErr:    "",
+		},
+		{
+			desc:           "empty arguments",
+			overrideParams: "",
+			expectedMap:    nil,
+			expectedErr:    "",
+		},
+		{
+			desc:           "zero value arugment",
+			overrideParams: "zero-value-arg=",
+			expectedMap:    map[string]string{"zero-value-arg": ""},
+			expectedErr:    "",
+		},
+		{
+			// TODO: Multiple arg values separated by , are not supported yet
+			desc:           "multiple equals characters in an argument",
+			overrideParams: "multiple-equalto-char=first-key=1",
+			expectedMap:    map[string]string{"multiple-equalto-char": "first-key=1"},
+			expectedErr:    "",
+		},
+		{
+			desc:           "incorrectly formatted argument",
+			overrideParams: "wrong-format-arg",
+			expectedErr:    "wrong format for override arg: wrong-format-arg",
+		},
+		{
+			desc:           "incorrectly formatted argument with only a value",
+			overrideParams: "=wrong-format-only-value",
+			expectedErr:    "wrong format for override arg: =wrong-format-only-value, arg name cannot be empty",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			options := aggregatedClusterRegistryOptions{
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerOverridesString: tc.overrideParams}}
+			err := options.MarshalOptions()
+			if tc.expectedErr == "" {
+				got := options.APIServerOverrides
+				want := tc.expectedMap
+
+				if !reflect.DeepEqual(got, want) {
+					t.Errorf("unexpected output: got: %v, want: %v", got, want)
+				}
+			} else {
+				if err.Error() != tc.expectedErr {
+					t.Errorf("unexpected error output: got: %s, want: %s", err.Error(), tc.expectedErr)
+				}
+			}
+		})
+	}
+}
+
+func TestCreateNamespace(t *testing.T) {
```

Comment:
Same here: this is already being tested in the `standalone` tests. It feels to me like a cleaner path towards the final goal is to add a few tests for the `aggregated`-specific functionality, refactor the `standlone` tests into `common`, and then flesh out the tests for `aggregated` and `standalone` specific features.

Reply:
That makes sense. I wanted to get something simple up for a start to get some ideas. I'm going to apply the [WIP] tag and work on making RBAC and APIService tests for aggregated first.

Reply:
That makes total sense to me. Looking forward to seeing the tests that you come up with!